multiwallet.go

```diff
@@ -0,0 +1,428 @@
+}
+
+func (mw *MultiWallet) LoadedWalletsCount() int32 {
+	return int32(len(mw.wallets))
+}
+
+func (mw *MultiWallet) OpenedWalletsRaw() []int {
+	wallets := make([]int, 0)
+	for _, w := range mw.wallets {
+		if w.WalletOpened() {
+			wallets = append(wallets, w.wallet.ID)
+		}
+	}
+
+	return wallets
+}
+
+func (mw *MultiWallet) OpenedWallets() string {
+	wallets := mw.OpenedWalletsRaw()
+	jsonEncoded, _ := json.Marshal(&wallets)
+
+	return string(jsonEncoded)
+}
+
+func (mw *MultiWallet) OpenedWalletsCount() int32 {
+	return int32(len(mw.OpenedWalletsRaw()))
+}
+
+func (mw *MultiWallet) SyncedWalletCount() int32 {
+	var syncedWallet int32
+	for _, w := range mw.wallets {
+		if w.WalletOpened() && w.synced {
+			syncedWallet++
+		}
+	}
+
+	return syncedWallet
+}
+
+func (mw *MultiWallet) CreateNewWallet(privatePassphrase string, spendingPassphraseType int32) (*LibWallet, error) {
+
+	if mw.activeSyncData != nil {
+		return nil, errors.New(ErrSyncAlreadyInProgress)
+	}
+
+	seed, err := GenerateSeed()
+	if err != nil {
+		return nil, err
+	}
+
+	properties := &Wallet{
+		Seed:                   seed,
+		SpendingPassphraseType: spendingPassphraseType,
+		DiscoveredAccounts:     true,
+	}
+
+	return mw.createWallet(properties, seed, privatePassphrase)
+}
+
+func (mw *MultiWallet) CreateWatchOnlyWallet(walletName string, extendedPublicKey string) (*LibWallet, error) {
+
+	exists, err := mw.WalletNameExists(walletName)
+	if err != nil {
+		return nil, err
+	} else if exists {
+		return nil, errors.New(ErrWalletNotLoaded)
+	}
+
+	w := &Wallet{
+		Name:               walletName,
+		DiscoveredAccounts: true,
+	}
+
+	err = mw.db.Save(w)
+	if err != nil {
+		return nil, err
+	}
+
+	homeDir := filepath.Join(mw.rootDir, strconv.Itoa(w.ID))
+	os.MkdirAll(homeDir, os.ModePerm) // create wallet dir
+
+	w.DataDir = homeDir
+	err = mw.db.Save(w) // update database with complete wallet information
+	if err != nil {
+		return nil, err
+	}
+
+	// delete from database if not created successfully
+	defer func() {
+		if err != nil {
+			mw.db.DeleteStruct(w)
+		}
+	}()
+
+	w.walletDbDriver = mw.dbDriver
+	w.netType = mw.activeNet.Name
+	libWallet, err := NewLibWallet(w)
+	if err != nil {
+		return nil, err
+	}
+
+	mw.wallets[w.ID] = libWallet
+
+	err = libWallet.CreateWatchingOnlyWallet(wallet.InsecurePubPassphrase, extendedPublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	go mw.listenForTransactions(libWallet)
+
+	return libWallet, nil
+}
+
+func (mw *MultiWallet) RestoreWallet(seedMnemonic, privatePassphrase string, spendingPassphraseType int32) (*LibWallet, error) {
+	if mw.activeSyncData != nil {
+		return nil, errors.New(ErrSyncAlreadyInProgress)
+	}
+
+	properties := &Wallet{
+		SpendingPassphraseType: spendingPassphraseType,
+		DiscoveredAccounts:     false,
+	}
+
+	return mw.createWallet(properties, seedMnemonic, privatePassphrase)
+}
+
+func (mw *MultiWallet) createWallet(w *Wallet, seedMnemonic, privatePassphrase string) (*LibWallet, error) {
```

Comment:
Modify this to accept `seed string, privatePassphrase string, spendingPassphraseType int32, discoveredAccounts bool` instead of `*Wallet` to eliminate the possibility of passing in an already created wallet.

Reply:
I don't think there's a possibility of passing an already created wallet.